src/streamGraphSettingsModel.ts

```diff
@@ -0,0 +1,212 @@
+import powerbiVisualsApi from "powerbi-visuals-api";
+import { formattingSettings } from "powerbi-visuals-utils-formattingmodel";
+import { legendInterfaces } from "powerbi-visuals-utils-chartutils";
+import LegendPosition = legendInterfaces.LegendPosition;
+
+import Card = formattingSettings.Card;
+import Model = formattingSettings.Model;
+
+import IEnumMember = powerbi.IEnumMember;
+
+class EnableGeneralCardSettings extends Card {
+    wiggle = new formattingSettings.ToggleSwitch({
+        name: "wiggle",
+        displayName: "Enable Wiggle",
+        displayNameKey: "Visual_Wiggle",
+        value: true,
+        topLevelToggle: false
+    });
+
+    name: string = "general";
+    displayName: string = "General";
+    displayNameKey: string = "Visual_General";
+    slices = [this.wiggle];
+}
+
+class BaseLabelColorCardSetting extends Card{
+    labelColor = new formattingSettings.ColorPicker({
+        name: "labelColor",
+        displayNameKey: "Visual_LabelsFill",
+        displayName: "Color",
+        value: { value: "#888888" }
+        // instanceKind: powerbi.VisualEnumerationInstanceKinds.ConstantOrRule
+    });
+}
+
+class BaseFontSizeCardSettings extends BaseLabelColorCardSetting{
+    fontSize = new formattingSettings.NumUpDown({
+        name: "fontSize",
+        displayName: "Text Size",
+        displayNameKey: "Visual_TextSize",
+        value: 12,
+        options: {
+            minValue: {
+                type: powerbiVisualsApi.visuals.ValidatorType.Min,
+                value: 8,
+            },
+            maxValue: {
+                type: powerbiVisualsApi.visuals.ValidatorType.Max,
+                value: 60,
+            }
+        }
+    });
+}
+
+class BaseAxisCardSettings extends BaseFontSizeCardSettings{
+    show = new formattingSettings.ToggleSwitch({
+        name: "show",
+        displayName: "show",
+        displayNameKey: "Visual_Show",
+        value: true,
+        topLevelToggle: true
+    });
+
+    showAxisTitle = new formattingSettings.ToggleSwitch({
+        name: "showAxisTitle",
+        displayNameKey: "Visual_Title",
+        displayName: "Title",
+        value: true,
+        topLevelToggle: false
+    });
+}
+
+const positionOptions : IEnumMember[] = [
+    {value : LegendPosition[LegendPosition.Top], displayName : "Top"}, 
```

Comment:
how is this connected to string resources?

Reply:
these are strings that will be used for dropdown translation